### PR TITLE
Add test view for touch-input-test

### PR DESCRIPTION
**Pull request by lin-erik**
This PR adds a `touch-input-view` for use in the `touch-input-test` to validate touch input. The view will be attached to the test, and a touch event will be injected. The test asserts that a correct response is sent back from the view (the location of the touch as well as the name of the view).

### CODE REVIEW

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+For example, if we inject at (-500, -500) `touch-input-view` will see a touch event at the middle of the upper-left quadrant of the screen.
+
+## Running the Test
+Start a Fuchsia package server
+```shell
+cd "$FUCHSIA_DIR"
+fx serve
+```
+
+Start the Fuchsia emulator in a graphical environment
```
Comment (akbiggs) [inline shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:44]: Just curious, any reason not to have run_integration-test.sh handle this by dropping `--skip-fuchsia-emu` from the next command?

More generally should we point this section at https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/tests/integration/README.md to avoid maintaining these instructions in two places?

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+Run the integration test
+```shell
+$ENGINE_DIR/flutter/tools/fuchsia/devshell/run_integration_test.sh touch-input --no-lto --skip-fuchsia-emu
+```
+
+## Playing around with `touch-input-view`
+
+This requires `tiles-session` (//src/session/examples/tiles-session), a simple graphical session with basic window management functionality.
```
Comment (akbiggs) [inline shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:56]: Just curious why doesn't this work with workstation's session (i.e. why is switching to tiles-session necessary?)

**Review by akbiggs — APPROVED** [review]
Wow this is awesome for two reasons:

1. Expanding our integration test coverage.
2. It's so well-documented, I learned a bunch from reading through it.

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md (hunk 2 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:56]: Running `ffx launch ...` without `tiles-session` resulted in nothing happening (followed [this documentation](https://fuchsia.dev/fuchsia-src/development/sessions/building-and-running-a-session) with `workstatation_eng.qemu-x64`)

Is workstation session something specific?

Comment (akbiggs) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:56]: workstation_session is a session that workstation boots up by default that you can use to run graphical components. Does `ffx session add` work (without `launch` beforehand)?

(note: you need to get past the "set your password" screen first)

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md (hunk 1 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:44]: The script defaults to launching a `--headless` emulator, which causes the test to fail.. I can add a flag to launch a graphical emulator (if this is what you meant by having the script handle it)

Good point on the documentation, I can update the README

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md (hunk 2 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:56]: That might be why it's failing, let me try again after setting my password

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md (hunk 1 above)
Comment (akbiggs) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md:44]: Oh shoot, I forgot about that. The Fuchsia tools tend to be graphical by default and support passing `--headless` for headless environments, so it's poor design that I didn't do the same.

We can also detect if we're in a headless environment with `[[ -z $DISPLAY ]]` I believe and use that to configure the emulator appropriately, but it might be better to be more explicit (graphical by default, pass `--headless` for non-graphics). We can handle this in a follow-up PR though.

File: shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md (hunk 2 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/touch-input/README.md]: It works!
